### PR TITLE
fix(SupportWidget): trigger query HasDonated when the user is logged in

### PR DESCRIPTION
**Pull request by pitb2022**

### CODE REVIEW

File: src/views/ArticleDetail/SupportWidget/index.tsx [hunk 1]
```diff
@@ -50,21 +50,30 @@ const SupportWidget = ({ article }: DonationProps) => {
...
+  useEffect(() => {
+    if (viewer.id !== '') {
+      getHasDonated({
+        variables: {
+          id: article.id,
+          senderId: viewer.id,
+        },
+      })
+    }
+  }, [viewer])
```
Comment (robertu7) [inline src/views/ArticleDetail/SupportWidget/index.tsx:69]: use `viewer.id`

File: src/views/ArticleDetail/SupportWidget/index.tsx [hunk 2]
```diff
@@ -50,21 +50,30 @@ const SupportWidget = ({ article }: DonationProps) => {
...
+  const [
+    getHasDonated,
+    { data: hasDonatedData, loading, refetch: hasDonatedRefetch },
+  ] = useLazyQuery<HasDonated>(HAS_DONATED, {
     fetchPolicy: 'network-only',
-    variables: { id: article.id, senderId: viewer.id },
   })
 
+  useEffect(() => {
+    if (viewer.id !== '') {
```
Comment (robertu7) [inline src/views/ArticleDetail/SupportWidget/index.tsx:61]: return-early

```
if (!viewer.id) return 

...
```

File: src/views/ArticleDetail/SupportWidget/index.tsx [hunk 3]
```diff
@@ -50,21 +50,30 @@ const SupportWidget = ({ article }: DonationProps) => {
...
+    }
+  }, [viewer])
+
   const hasDonatedArticle =
     hasDonatedData?.article as HasDonated_article_Article
 
   useEffect(() => {
     if (hasDonatedData) {
-      if (viewer.id !== '' && hasDonatedArticle?.donation.totalCount === 1) {
+      if (hasDonatedArticle?.donation.totalCount === 1) {
```
Comment (robertu7) [inline src/views/ArticleDetail/SupportWidget/index.tsx:76]: `hasDonatedArticle?.donation.totalCount >= 1`?

**Review by robertu7 — CHANGES_REQUESTED** [review]

File: src/views/ArticleDetail/SupportWidget/index.tsx (hunk 3 above)
Comment (pitb2022) [reply src/views/ArticleDetail/SupportWidget/index.tsx:76]: This request is only for the current user, so it must be equal to one

**Review by robertu7 — APPROVED** [review]

**Pull request merged**